lib/passports.js

```diff
@@ -101,14 +101,26 @@ module.exports = {
                 passport.tokens = JSON.stringify(query.tokens)
               }
 
-              // Save any updates to the Passport before moving on
-              app.services.FootprintService.update('Passport', passport.id, passport) .then(passport => {
-                app.services.FootprintService.find('user', passport[0].user.id)
-                  .then(user => {
-                    next(null, user)
-                  })
-                  .catch(next)
-              }).catch(next)
+             // TODO: Must be tested for check if ok for both Sequelize and Waterline
+             // Save any updates to the Passport before moving on
+              app.services.FootprintService.find('Passport', passport.id)
```

Comment:
Why searching for a new passport instance  when you already have it ? The find is few lines above so if you have it here no need to do another search just if(passport) should be enough